redis/runtime/src/main/java/io/quarkiverse/langchain4j/redis/RedisEmbeddingStore.java

```diff
@@ -0,0 +1,246 @@
+            ReactiveTransactionalJsonCommands<String> json = ds.json();
+            int size = ids.size();
+            Uni[] unis = new Uni[size];
+            for (int i = 0; i < size; i++) {
+                String id = ids.get(i);
+                Embedding embedding = embeddings.get(i);
+                TextSegment textSegment = embedded == null ? null : embedded.get(i);
+                Map<String, Object> fields = new HashMap<>();
+                fields.put(schema.getVectorFieldName(), embedding.vector());
+                if (textSegment != null) {
+                    fields.put(schema.getScalarFieldName(), textSegment.text());
+                    fields.putAll(textSegment.metadata().asMap());
+                }
+                String key = schema.getPrefix() + id;
+                unis[i] = json.jsonSet(key, "$", fields);
+            }
+            return Uni.join().all(unis).andFailFast().replaceWithVoid();
+        }).await().indefinitely();
+    }
+
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults,
+            double minScore) {
+        String queryTemplate = "*=>[ KNN %d @%s $BLOB AS %s ]";
+        String query = format(queryTemplate, maxResults, schema.getVectorFieldName(), SCORE_FIELD_NAME);
+        // TODO: rewrite to the data source api, but we need a new
+        // method QueryArgs.param(String, byte[]) to get it working
+
+        //        QueryArgs args = new QueryArgs()
+        //                .sortByAscending(SCORE_FIELD_NAME)
+        //                .param("DIALECT", "2")
+        //                .param("BLOB", toByteArray(referenceEmbedding.vector()));
+        //        Uni<SearchQueryResponse> search = ds.search()
+        //                .ftSearch(schema.getIndexName(), query, args);
+        //        SearchQueryResponse response = search.await().indefinitely();
+        Request request = Request.cmd(Command.FT_SEARCH)
+                .arg(schema.getIndexName())
+                .arg(query)
+                .arg("PARAMS")
+                .arg("2")
+                .arg("BLOB")
+                .arg(toByteArray(referenceEmbedding.vector()))
+                .arg("DIALECT")
+                .arg("2");
+        Response response = ds.getRedis().send(request).await().indefinitely();
+        return StreamSupport.stream(response.get("results").spliterator(), false)
+                .map(this::toEmbeddingMatch)
+                .filter(embeddingMatch -> embeddingMatch.score() >= minScore)
+                .collect(toList());
+    }
+
+    /**
+     * Deletes all keys with the prefix that is used by this embedding store.
+     */
+    public void deleteAll() {
+        Response response = ds.execute("SCAN", "0", "MATCH", schema.getPrefix() + "*").await().indefinitely();
+        Set<String> keysToDelete = new TreeSet<>();
+        for (Response response1 : response.get(1)) {
+            keysToDelete.add(response1.toString());
```

Comment:
That's not how you should use cursors.
Scan won't return all the keys, but a cursor to iterate over the key. Use the data source instead (it does the "right" thing).